connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConfig.java

```diff
@@ -73,6 +75,19 @@ public class MirrorSourceConfig extends MirrorConnectorConfig {
     public static final String SYNC_TOPIC_CONFIGS_INTERVAL_SECONDS = SYNC_TOPIC_CONFIGS + INTERVAL_SECONDS_SUFFIX;
     private static final String SYNC_TOPIC_CONFIGS_INTERVAL_SECONDS_DOC = "Frequency of topic config sync.";
     public static final long SYNC_TOPIC_CONFIGS_INTERVAL_SECONDS_DEFAULT = 10 * 60;
+    @Deprecated
+    public static final String USE_INCREMENTAL_ALTER_CONFIG = "use.incremental.alter.configs";
+    private static final String USE_INCREMENTAL_ALTER_CONFIG_DOC = "Deprecated. Which API to use for syncing topic configs. " +
+            "The valid values are \"requested\", \"required\" and \"never\". " +
+            "By default, set to \"requested\", which means the IncrementalAlterConfigs API is being used for syncing topic configurations " +
+            "and if any request receives an error from an incompatible broker, it will fallback to using the deprecated AlterConfigs API. " +
+            "If explicitly set to \"required\", the IncrementalAlterConfigs API is used without the fallback logic and +" +
+            "if it receives an error from an incompatible broker, the connector will fail." +
+            "If explicitly set to \"never\", the AlterConfig is always used." +
+            "This setting will be removed and the behaviour of \"required\" will be used in Kafka 4.0, therefore users should ensure that target broker is at least 2.3.0";
```

Comment:
Nit: it might be easier/cleaner to use single quotes around property names instead of double quotes, so that we don't have to use escape sequences.

```suggestion
            "The valid values are 'requested', 'required' and 'never'. " +
            "By default, set to 'requested', which means the IncrementalAlterConfigs API is being used for syncing topic configurations " +
            "and if any request receives an error from an incompatible broker, it will fallback to using the deprecated AlterConfigs API. " +
            "If explicitly set to 'required', the IncrementalAlterConfigs API is used without the fallback logic and +" +
            "if it receives an error from an incompatible broker, the connector will fail." +
            "If explicitly set to 'never', the AlterConfig is always used." +
            "This setting will be removed and the behaviour of 'required' will be used in Kafka 4.0, therefore users should ensure that target broker is at least 2.3.0";
```